qiskit/primitives/base/base_result.py

```diff
@@ -71,17 +76,38 @@ def _generate_experiments(self) -> Iterator[ExperimentData]:
         for values in zip(*self._field_values):
             yield dict(zip(names, values))
 
+    @deprecate_func(since="0.46.0")
     def decompose(self) -> Iterator[BasePrimitiveResult]:
         """Generate single experiment result objects from self."""
         for values in zip(*self._field_values):
             yield self.__class__(*[(v,) for v in values])
 
-    @property  # TODO: functools.cached_property when py37 is droppped
+    @property  # TODO: functools.cached_property when py37 is dropped
     def _field_names(self) -> tuple[str, ...]:
         """Tuple of field names in any inheriting result dataclass."""
         return tuple(field.name for field in fields(self))
 
-    @property  # TODO: functools.cached_property when py37 is droppped
+    @property  # TODO: functools.cached_property when py37 is dropped
     def _field_values(self) -> tuple:
         """Tuple of field values in any inheriting result dataclass."""
         return tuple(getattr(self, name) for name in self._field_names)
+
+
+class BasePrimitiveResult(_BasePrimitiveResult):
+    """DEPRECATED: Primitive result abstract base class.
+
+    Base class for Primitive results meant to provide common functionality to all inheriting
+    result dataclasses.
+    """
+
+    # Override to add deprecation warning if a user implements this class directly
+
+    def __post_init__(self) -> None:
+        warnings.warn(
+            "The BasePrimitiveResult class is deprecated as of Qiskit 0.46"
+            " and will be removed or heavily modified in Qiskit 1.0. For now"
+            " use EstimatorResult or SamplerResult directly.",
+            category=DeprecationWarning,
+            stacklevel=2,
+        )
+        super().__post_init__()
```

Comment:
Neither this class nor anything in its inheritance chain is a `dataclasses.dataclass`, so the `__post_init__` method will not get called if it is instantiated.  You might be better off deprecating retrieving the `BasePrimitiveResult` _name_ from all modules that it's publicly exported from.

Reply:
I implemented @jakelishman suggestion to deprecate on attempting to import BasePrimitiveResult